pyfr/integrators/dual/pseudo/pseudosteppers.py

```diff
@@ -337,3 +337,316 @@ class DualRK45PseudoStepper(DualRKVdH2RPseudoStepper):
+    ]
+
+    c = [
+        0, 0.227510900982561, 0.4321965004909342, 0.659707401473495,
+        0.887218302456056, 1.065544162796411, 1.014336438253882
+    ]
+
+
+class DualOptRKp6(DualOptRKPseudoStepper):
+    pseudo_stepper_name = 'optrk6'
+
+    a = [
+        [0, 0, 0, 0, 0, 0, 0],
+        [0.2231960555901207, 0, 0, 0, 0, 0, 0],
+        [0.1689692883853581, 0.1689692883853581, 0, 0, 0, 0, 0],
+        [0.1663133174662472, 0.166313317464247, 0.2196877125118087, 0, 0, 0,
+         0],
+        [0.1749219945384915, 0.1413513027445673, 0.1864400061578061,
+         0.1894173939126585, 0, 0, 0],
+        [0.1749242273964011, 0.1414254100902069, 0.1861307578969414,
+         0.1890118533187293, 0.2227181952464798, 0, 0],
+        [0.1749239374908155, 0.1414251882006011, 0.1861304661814622,
+         0.189011557128436, 0.2227177430366584, 0.2231956024100457, 0]
+    ]
+
+    b = [
+        0.1679668867629793, 0.134622771270711, 0.1637470513385491,
+        0.143194963257529, 0.1599767054621269, 0.1152456939556625,
+        0.1152459279524424
+    ]
+
+    c = [
+        0, 0.2231960555901207, 0.3379385767707163, 0.5523143474423029,
+        0.6921306973535233, 0.9142104439487584, 1.137404494448019
+    ]
+
+
+class DualOptRKp7(DualOptRKPseudoStepper):
+    pseudo_stepper_name = 'optrk7'
+
+    a = [
+        [0, 0, 0, 0, 0, 0, 0],
+        [0.2194110578498075, 0, 0, 0, 0, 0, 0],
+        [0.1466669309147015, 0.1466669309147015, 0, 0, 0, 0, 0],
+        [0.1511367650235299, 0.1376548377319136, 0.205929130558191, 0, 0, 0,
+         0],
+        [0.1520469747214135, 0.1357064919104657, 0.2028574104512142,
+         0.2161382359995746, 0, 0, 0],
+        [0.152704409014835, 0.1353452840943641, 0.2008657130837393,
+         0.2138955841791913, 0.2171344472074404, 0, 0],
+        [0.1541281616423111, 0.1331010373841963, 0.1965785379405185,
+         0.209330306125214, 0.2125000405159577, 0.2147280603440595, 0]
+    ]
+
+    b = [
+        0.1442647477912376, 0.1263213794181965, 0.1833551644876998,
+        0.1624709034365959, 0.1316909975417381, 0.1245898159757726,
+        0.1273069913487596
+    ]
+
+    c = [
+        0, 0.2194110578498075, 0.2933338618294031, 0.4947207333136346,
+        0.706749113082668, 0.9199454375795701, 1.120366143952257
+    ]
+
+
+class DualOptRKp8(DualOptRKPseudoStepper):
+    pseudo_stepper_name = 'optrk8'
+
+    a = [
+        [0, 0, 0, 0, 0, 0, 0],
+        [0.2161768331428258, 0, 0, 0, 0, 0, 0],
+        [0.2161763099027749, 0.2161763099027749, 0, 0, 0, 0, 0],
+        [0.216173556224235, 0.216173556224235, 0.2161740794576209, 0, 0, 0, 0],
+        [0.2161735562337322, 0.2161735562337322, 0.2161734529385136,
+         0.2161762066157379, 0, 0, 0],
+        [0.2161732559391219, 0.2161732559391219, 0.2161728401742457,
+         0.2161755938436643, 0.2161762203689764, 0, 0],
+        [0.2161732559425067, 0.2161732559425067, 0.2161728401882148,
+         0.2161755938576336, 0.2161737836501301, 0.2161743964170724, 0]
```

Comment:
These numbers look dirty.  Are you sure they are not (in reality) all supposed to be the same value?

Reply:
The Butcher tableaus are generated from the optimised stability polynomial coefficients. For a given stability polynomial, the Butcher tableau is under defined. I have plotted the stability regions of these tableaus and they should be fine.

Reply:
Okay.  What is typically done in the literature here is for rational coefficients to be employed (see Carpenter and Kennedy for instance).